src/System.Console/src/System/TermInfo.cs

```diff
@@ -208,20 +209,20 @@ private static Database ReadDatabase(string term)
             /// <param name="filePath">The path to the file to open.</param>
             /// <param name="fd">If successful, the opened file descriptor; otherwise, -1.</param>
             /// <returns>true if the file was successfully opened; otherwise, false.</returns>
-            private static bool TryOpen(string filePath, out int fd)
+            private static bool TryOpen(string filePath, out SafeFileHandle fd)
             {
-                int tmpFd;
-                while ((tmpFd = Interop.Sys.Open(filePath, Interop.Sys.OpenFlags.O_RDONLY, 0)) < 0)
+                IntPtr tmpFd;
+                while ((int)(tmpFd = Interop.Sys.Open(filePath, Interop.Sys.OpenFlags.O_RDONLY, 0)) < 0)
```

Comment:
Nit: this shouldn't make any difference, but just in the name of robustness, should this cast be to long rather than int?